libraries/AP_Proximity/AP_Proximity_LightWareSF40C.cpp

```diff
@@ -414,16 +421,22 @@ void AP_Proximity_LightWareSF40C::process_message()
                             _distance_valid[last_sector] = false;
                         }
                         // update boundary used for avoidance
-                        update_boundary_for_sector(last_sector, true);
+                        update_boundary_for_sector(last_sector, false);
                     }
                     // initialise for new sector
                     last_sector = sector;
                     last_distance_cm = INT16_MAX;
                 }
-                // use shortest valid distance
-                if ((dist_cm < last_distance_cm) && (dist_cm >= dist_min_cm) && (dist_cm <= dist_max_cm)) {
-                    last_distance_cm = dist_cm;
-                    last_angle_deg = angle_deg;
+                if ((dist_cm >= dist_min_cm) && (dist_cm <= dist_max_cm)) {
+                    // use shortest valid distance for this sector's distance
+                    if (dist_cm < last_distance_cm) {
+                        last_distance_cm = dist_cm;
+                        last_angle_deg = angle_deg;
+                    }
+                    // send point to object avoidance database
+                    if (database_ready) {
+                        database_push(angle_deg, last_distance_cm * 0.01f, last_distance_received_ms, current_loc, current_vehicle_bearing);
```

Comment:
This is pushing this point angle with the smallest distance for the sector. Is that correct?

Reply:
thanks for this, no, this is definitely not correct actually.  We want it sending all the distances it's received not just the shortest, fixed!